test/runtime/Policies.go

```diff
@@ -897,9 +899,24 @@ var _ = Describe("RuntimePolicies", func() {
     ]
   }
 ]`
-		_, err := vm.PolicyRenderAndImport(fqdnPolicy)
+		startPolicyRevision, err := vm.PolicyGetRevision()
+		Expect(err).To(BeNil(), "Unable to get start policy revision before policy import", err)
+		_, err = vm.PolicyRenderAndImport(fqdnPolicy)
 		Expect(err).To(BeNil(), "Unable to import policy: %s", err)
 
+		// The DNS poll will update the policy and regenerate. Since this is the
+		// first ever update, we can simply check for the presence of toCIDRSet in
+		// the policy.
+		// Once we have an API to expose DNS->IP mappings we can also use that to
+		// ensure the lookup has completed more explicitly
+		timeout_s := 3 * fqdn.DNSPollerInterval / time.Second // convert to seconds
+		dnsWaitBody := func() bool {
+			return vm.PolicyWait(startPolicyRevision + 2).WasSuccessful()
```

Comment:
This seems really hacky to me. Why can't we query the policy repository itself to wait for toCIDRSets within the policy rules to be updated? You imply this above, but then we simply wait for the revision number to increase by 2.

Reply:
We can. It seemed like a wash to me. If I get the policy, I would have to parse it (it's json, I think) or simply search for the presence of a `ToCIDRSet` section. In the case where I parse it, I end up doing the same thing as the substring search (I don't know the IPs beforehand, so I can't check those). In both cases I would have to assume this test is the only one manipulating the policy repository.
In the case where I look at revisions, I know that there will be two additions to the policy repository, and so the revision would increment twice. I also have to assume that only this test is manipulating the policy repository, since any other modification would be bad and increment the policy revision.

I did, in fact, try the substring search method. It worked but seemed a little random, especially the string search I was doing.

Reply:
ok, we allow for setting an "id" on the policy, so I can grab the exact same one back. This is what it would look like. What do you think?
```golang
        It("Enforces ToFQDNs policy", func() {
                By("Importing policy with ToFQDN rules")
                // notaname.cilium.io never returns IPs, and is there to test that the
                // other name does get populated.
                testPolicyID := rand.Int63n(1000)
                testPolicyIDString := fmt.Sprintf("toFQDNS-test-label=%d", testPolicyID)
                fqdnPolicy := fmt.Sprintf(`
[
  {
    "labels": [{
                        "key": "toFQDNS-test-label",
                        "value": "%d"
                }],
    "endpointSelector": {
      "matchLabels": {
        "container:id.app1": ""
      }
    },
    "egress": [
      {
        "toPorts": [{
          "ports":[{"port": "53", "protocol": "ANY"}]
        }]
      },
      {
        "toFQDNs": [
          {
            "matchName": "cilium.io"
          },
          {
            "matchName": "notaname.cilium.io"
          }
        ]
      }
    ]
  }
]`, testPolicyID)

                _, err := vm.PolicyRenderAndImport(fqdnPolicy)
                Expect(err).To(BeNil(), "Unable to import policy: %s", err)
                defer vm.PolicyDel(testPolicyIDString)

                // The DNS poll will update the policy and regenerate. Since this is the
                // first ever update, we can simply check for the presence of toCIDRSet in
                // the policy, and that the policy revision has incremented twice: once for
                // the policy insert, and another for the DNS IP updates to toCIDRSet.
                // Once we have an API to expose DNS->IP mappings we can also use that to
                // ensure the lookup has completed more explicitly
                timeout_s := 3 * fqdn.DNSPollerInterval / time.Second // convert to seconds
                dnsWaitBody := func() bool {
                        policyResult := vm.PolicyGet(testPolicyIDString)
                        if !policyResult.WasSuccessful() {
                                return false
                        }
                        policy := policyResult.GetStdOut()
                        return strings.Contains(policy, "toFQDNs") && strings.Contains(policy, "toCIDRSet")
                }
                err = helpers.WithTimeout(dnsWaitBody, "DNSPoller did not update IPs",
                        &helpers.TimeoutConfig{Ticker: 1, Timeout: timeout_s})

                Expect(vm.WaitEndpointsReady()).Should(BeTrue(), "Endpoints are not ready after ToFQDNs DNS poll triggered a regenerate")

                By("Denying egress to IPs of DNS names not in ToFQDNs, and normal IPs")
                // www.cilium.io has a different IP than cilium.io (it is CNAMEd as well!),
                // and so should be blocked.
                // cilium.io.cilium.io doesn't exist.
                // 1.1.1.1, amusingly, serves HTTP.
                for _, blockedTarget := range []string{"www.cilium.io", "cilium.io.cilium.io", "1.1.1.1"} {
                        res := vm.ContainerExec(helpers.App1, helpers.CurlFail(blockedTarget))
                        res.ExpectFail("Curl succeeded against blocked DNS name %s" + blockedTarget)
                }

                By("Allowing egress to IPs of specified ToFQDN DNS names")
                allowedTarget := "cilium.io"
                res := vm.ContainerExec(helpers.App1, helpers.CurlWithHTTPCode(allowedTarget))
                res.ExpectContains("301", "Cannot access %s %s", allowedTarget, res.OutputPrettyPrint())
        })
```